ee/clickhouse/queries/funnels/funnel_correlation_persons.py

```diff
@@ -1,144 +1,189 @@
 from posthog.models.team import Team
 
 
-class FunnelCorrelationPersons:
+class FunnelCorrelationActors:
     def __init__(self, filter: Filter, team: Team, base_uri: str = "/") -> None:
-        self._funnel_correlation = FunnelCorrelation(filter, team, base_uri=base_uri)
+        self._base_uri = base_uri
         self._filter = filter
         self._team = team
 
         if not self._filter.correlation_person_limit:
             self._filter = self._filter.with_data({FUNNEL_CORRELATION_PERSON_LIMIT: 100})
 
-    def run(self):
-        """
-        Returns `ReturnList` type, generated by `serializers.serialize`, which returns the Person model.
-        """
+    def get_actors(
+        self,
+    ) -> Tuple[Union[QuerySet[Person], QuerySet[Group]], Union[List[SerializedGroup], List[SerializedPerson]]]:
+        if self._filter.correlation_type == FunnelCorrelationType.PROPERTIES:
+            return _FunnelPropertyCorrelationActors(self._filter, self._team, self._base_uri).get_actors()
+        else:
+            return _FunnelEventsCorrelationActors(self._filter, self._team, self._base_uri).get_actors()
 
-        query, params = self.get_query()
-        results: List[Tuple[str]] = sync_execute(query, params)
 
-        return self._format_results(results)
+class _FunnelEventsCorrelationActors(ActorBaseQuery):
+    def __init__(self, filter: Filter, team: Team, base_uri: str = "/") -> None:
+        self._funnel_correlation = FunnelCorrelation(filter, team, base_uri=base_uri)
+        super().__init__(team, filter)
 
-    def get_query(self) -> Tuple[str, Dict[str, Any]]:
-        if self._filter.correlation_type == FunnelCorrelationType.PROPERTIES:
-            return self.get_properties_query()
+    @cached_property
+    def is_aggregating_by_groups(self) -> bool:
+        return self._filter.aggregation_group_type_index is not None
 
-        else:
-            return self.get_events_query()
+    def actor_query(self, extra_fields: Optional[List[str]] = None):
 
-    def get_properties_query(self) -> Tuple[str, Dict[str, Any]]:
+        if not self._filter.correlation_person_entity:
+            raise ValidationError("No entity for persons specified")
 
-        if not self._filter.correlation_property_values:
-            raise ValidationError("Property Correlation expects atleast one Property to get persons for")
+        assert isinstance(self._filter.correlation_person_entity, Entity)
 
         funnel_persons_query, funnel_persons_params = self._funnel_correlation.get_funnel_persons_cte()
 
+        prop_filters = self._filter.correlation_person_entity.properties
+        prop_query, prop_params = FunnelEventQuery(self._filter, self._team.pk)._get_props(prop_filters)
+
         conversion_filter = (
-            f'funnel_people.steps {"=" if self._filter.correlation_persons_converted else "<>"} target_step'
+            f'AND person.steps {"=" if self._filter.correlation_persons_converted else "<>"} target_step'
```

Comment:
This particular change is confusing for me - looking below, where is the `person` table coming from? Shouldn't this be funnel_people?